src/main/webapp/app/lecture/manage/lecture-units/create-attachment-video-unit/create-attachment-video-unit.component.ts

```diff
@@ -83,6 +84,15 @@ export class CreateAttachmentVideoUnitComponent implements OnInit {
             .pipe(
                 switchMap((response) => {
                     const lectureUnit = response.body!;
+
+                    // Trigger transcript generation if enabled
+                    if (generateTranscript && lectureUnit.id) {
```

Comment:
This could be a variable instead of a comment 😄 
```suggestion
                    const shouldTriggerTranscriptGeneration = generateTranscript && this.attachmentVideoUnit.id;
                    if (shouldTriggerTranscriptGeneration) {
```